File: 8 - Dictionaries and Maps/Solution.java
import java.util.*;
import java.io.*;

class Solution{
    public static void main(String []argh){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        
        Map<String,Integer> myMap = new HashMap<String,Integer>(); 
        
        for(int i = 0; i < n; i++){
            String name = in.next();
            int phone = in.nextInt();
            // Write code here
            myMap.put(name, phone);
        }
        while(in.hasNext()){
            String s = in.next();
            // Write code here
            if (myMap.get(s)!=null)                
                System.out.println(s + "=" + myMap.get(s) );
            else
                System.out.println("Not found");
        }
        in.close();
    }
}

File: 8 - Dictionaries and Maps/Solution.kt
import java.util.*

fun main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT. */
    val value = Scanner(System.`in`)

    val n = value.nextInt()
    val phoneBook = hashMapOf<String, Int>()

    for (i in 0 until n) {
        val name = value.next()
        val phone = value.nextInt()
        phoneBook[name] = phone
    }

    while (value.hasNext()) {
        val name = value.next()
        if (name in phoneBook) {
            val phone = phoneBook[name]
            println("$name=$phone")
        } else
            println("Not found")
    }

    value.close()
}
